contracts/lib/PolicyLib.sol

```diff
@@ -117,6 +117,10 @@ library PolicyLib {
             revert ISmartSession.InvalidSelfCall();
         }
 
+        // under no circumstances should the target be the smart session. This could allow session keys to elevate their
+        // privileges
+        if (target == address(this)) revert ISmartSession.InvalidCallTarget();
```

Comment:
@filmakarov would be interested to hear your opinion here.
I think sessions keys having authority to call smartsession would be very dangerous.